src/aleph/vm/pool.py

```diff
@@ -152,11 +159,21 @@ async def stop_vm(self, vm_hash: ItemHash) -> Optional[VmExecution]:
         """Stop a VM."""
         execution = self.executions.get(vm_hash)
         if execution:
-            await execution.stop()
+            if execution.persistent:
+                await self.stop_persistent_execution(execution)
+            else:
+                await execution.stop()
             return execution
         else:
             return None
 
+    async def stop_persistent_execution(self, execution):
+        """Stop persistent VMs in the pool."""
+        assert execution.persistent, "Execution isn't persistent"
+        self.systemd_manager.stop_and_disable(execution.controller_service)
+        await execution.stop()
+        execution.persistent = False
```

Comment:
What would happen if one of the two lines above would crash ?

Reply:
The same that happened before, I just added a method to use systemd instead of the old stop() method.
For the stop method on systemd, it is already controlled to not raise exceptions.